api/Service/EventService.cs

```diff
@@ -48,16 +49,30 @@ public async Task<Event> CreateEvent(EventDto eventDto)
 				Tags = eventDto.Tags,
 				BackgroundId = await _pictureService.CreatePictureByContent(eventDto.BackgroundImage)
 			};
+
+			var updateCategory = await _categoryService.GetCategory(eventDto.CategoryId);
+			updateCategory.HasEvents = true;
+			await _categoryService.PutCategory(eventDto.CategoryId, updateCategory);
+
```

Comment:
this should be a new function 

updateCategoryHasEvents(bool hasEvents) {
             var updateCategory = await _categoryService.GetCategory(eventDto.CategoryId);
			updateCategory.HasEvents = hasEvents;
			await _categoryService.PutCategory(eventDto.CategoryId, updateCategory);
}